### PR TITLE
Upgrade java, maven and versions of other dependencies

### PR DESCRIPTION
**Pull request by velinovskav**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -79,35 +79,35 @@
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
 
-    <source.version>1.7</source.version>
-    <target.version>1.7</target.version>
+    <source.version>9</source.version>
+    <target.version>9</target.version>
```
Comment (marcelstoer) [inline pom.xml]: This should still be runnable on Java 8.

Comment (velinovskav) [reply pom.xml]: Ok, then I will set source/target version to 1.8.

File: valdr-bean-validation/pom.xml [hunk 2]
```diff
@@ -48,6 +48,12 @@
       <artifactId>hibernate-validator</artifactId>
     </dependency>
 
+    <dependency>
+      <groupId>javax.xml.bind</groupId>
+      <artifactId>jaxb-api</artifactId>
+      <version>2.3.0</version>
+    </dependency>
```
Comment (marcelstoer) [inline valdr-bean-validation/pom.xml]: I haven't figured out yet what you need JAXB for - what did I miss?

**Comment by marcelstoer** [comment]
This is taking shape quite nicely. However, the more I think about it the more hesitant I become. There's a clearly defined mapping between Bean Validation constraints and valdr Angular JS validators as per https://github.com/netceteragroup/valdr-bean-validation#mapping-of-bean-validation-constraints-to-valdr-constraints (which this PR should also update).  This is established through a String match from e.g. https://github.com/netceteragroup/valdr/blob/master/src/core/validators/hibernateEmailValidator.js#L12 to https://github.com/netceteragroup/valdr-bean-validation/blob/master/valdr-bean-validation/src/main/java/com/github/valdr/BuiltInConstraint.java#L33. So, if you change `hibernateEmail` to `email` you'll effectively use a different validator (the one here https://github.com/netceteragroup/valdr-bean-validation/blob/master/valdr-bean-validation/src/main/java/com/github/valdr/BuiltInConstraint.java#L33).

I'm thinking that maybe we should add support for the Bean Validation `email` constraint without dropping the Hibernate email support at the same time. They could basically co-exist. However, we could also argue that anyone stuck with the proprietary and deprecated Hibernate stuff should simply not upgrade _this_ library.

What do you think?

**Pull request merged**